### PR TITLE
Add tests for Sorting/bogo_sort.rb and run tests on CI

### PR DESCRIPTION
**Pull request by vzvu3k6k**
- Adds simple test cases for Sorting/bogo_sort.rb using [minitest](https://github.com/seattlerb/minitest)
- Set up GitHub Actions to run these tests on the latest Ruby (v2.7)
    - To make adding tests easy, uses [`Rake::TestTask`](https://ruby.github.io/rake/Rake/TestTask.html)

### CODE REVIEW

File: DIRECTORY.md [hunk 1]
```diff
@@ -42,6 +42,7 @@
 
 ## Sorting
   * [Bogo Sort](https://github.com/TheAlgorithms/Ruby/blob/master/sorting/bogo_sort.rb)
+  * [Bogo Sort Test](https://github.com/TheAlgorithms/Ruby/blob/master/sorting/bogo_sort_test.rb)
```
Comment (vbrazo) [inline DIRECTORY.md:45]: Could we consider merging `Bogo Sort` + `Bogo Sort Test` in the same directory and have only 1 item in the `DIRECTORY.md`?

**Review by vbrazo — COMMENTED** [review]
That's awesome. I mentioned here https://github.com/TheAlgorithms/Ruby/issues/66 that we could possibly add a test suite + a linter for this repository and it looks like you had already worked on that.

Just left a minor suggestion to see your thoughts, but LGTM. Thanks!

File: DIRECTORY.md (hunk 1 above)
Comment (vbrazo) [reply DIRECTORY.md:45]: Checking the other repos, I noticed that we don't do this, so I'm just removing and merging. Thanks.

**Review by vbrazo — APPROVED** [review]

File: DIRECTORY.md (hunk 1 above)
Comment (vzvu3k6k) [reply DIRECTORY.md:45]: Thanks for comments.

It seems that GitHub Action has automatically added `Bogo Sort Test` again on 687840f22f6dab1a2b08395e8df39630099660f2. Maybe we should consider update [the workflow](https://github.com/TheAlgorithms/Ruby/blob/687840f22f6dab1a2b08395e8df39630099660f2/.github/workflows/update_directory_md.yml) to skip `*_test.rb` in building `DIRECTORY.md`.

Let me know if you have any feedback about testing (opening a new issue, probably).

**Pull request merged**